api/src/test/java/com/epam/pipeline/controller/region/CloudRegionControllerTest.java

```diff
@@ -0,0 +1,460 @@
+        gcpRegion = RegionCreatorUtils.getDefaultGcpRegion();
+
+        awsRegionDTO = RegionCreatorUtils.getDefaultAwsRegionDTO();
+        azureRegionDTO = RegionCreatorUtils.getDefaultAzureRegionDTO();
+        gcpRegionDTO = RegionCreatorUtils.getDefaultGcpRegionDTO();
+
+        expectedAwsRegionResult = ControllerTestUtils.buildExpectedResult(awsRegion);
+        expectedAzureResult = ControllerTestUtils.buildExpectedResult(azureRegion);
+        expectedGcpResult = ControllerTestUtils.buildExpectedResult(gcpRegion);
+    }
+
+    @Test
+    public void shouldFailLoadProvidersForUnauthorizedUser() throws Exception {
+        mvc().perform(get(LOAD_PROVIDERS_URL)
+                .servletPath(SERVLET_PATH))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadProviders() throws Exception {
+        final List<CloudProvider> cloudProviders = Collections.singletonList(CloudProvider.AWS);
+
+        Mockito.doReturn(cloudProviders).when(mockCloudRegionApiService).loadProviders();
+
+        final MvcResult mvcResult = mvc().perform(get(LOAD_PROVIDERS_URL)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(EXPECTED_CONTENT_TYPE))
+                .andReturn();
+
+        Mockito.verify(mockCloudRegionApiService).loadProviders();
+
+        final ResponseResult<List<CloudProvider>> expectedResult =
+                ControllerTestUtils.buildExpectedResult(cloudProviders);
+
+        ControllerTestUtils.assertResponse(mvcResult, getObjectMapper(), expectedResult,
+                new TypeReference<Result<List<CloudProvider>>>() { });
+    }
+
+    @Test
+    public void shouldFailLoadAllForUnauthorizedUser() throws Exception {
+        mvc().perform(get(REGION_URL)
+                .servletPath(SERVLET_PATH))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadAll() throws Exception {
+        final List<AwsRegion> cloudRegions = Collections.singletonList(awsRegion);
+
+        Mockito.doReturn(cloudRegions).when(mockCloudRegionApiService).loadAll();
+
+        final MvcResult mvcResult = mvc().perform(get(REGION_URL)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(EXPECTED_CONTENT_TYPE))
+                .andReturn();
+
+        Mockito.verify(mockCloudRegionApiService).loadAll();
+
+        final ResponseResult<List<AwsRegion>> expectedResult =
+                ControllerTestUtils.buildExpectedResult(cloudRegions);
+
+        ControllerTestUtils.assertResponse(mvcResult, getObjectMapper(), expectedResult,
+                new TypeReference<Result<List<AwsRegion>>>() { });
```

Comment:
Could we move it to class fields to reuse? (`new TypeReference<Result<List<AwsRegion>>>() { }`)